packages/api/src/beacon/server/events.ts

```diff
@@ -36,6 +37,9 @@ export function getRoutes(config: ChainForkConfig, api: ServerApi<Api>): ServerR
           await new Promise<void>((resolve, reject) => {
             void api.eventstream(req.query.topics, controller.signal, (event) => {
               try {
+                // If the request is already aborted, we don't need to send any more events.
+                if (req.raw.destroyed) return;
```

Comment:
This check will try to avoid headers already sent scenario.

Reply:
Are you sure this handles all cases where this can happen, I was think maybe to check `res.raw.headersSent` before setting hearders but would also have to look into this more if that the correct condition to check for

Reply:
headers already sent error occurred, when you try to write to a response, which related request is completed. That completion can be because request was aborted, or server already finished writing response and closed the stream with `res.end()`.  

I can't see the later case happens in our code, as we write the response only from one place. So the only case left in my mind is the first one, so this condition will cover it. 

If you can find a way to reproduce `headers already sent` scenario manually, please share then we can test.

Reply:
Based on the stack trace in the issue, the error is not caused by us setting headers but something fastify does.

Looks like here [res.writeHead]( https://github.com/fastify/fastify/blob/cc347d7c0b4266097b61b126158b797878668353/lib/reply.js#L597), might be an issue on fastify or related to us not closing correcty.

But `res.raw.write`  looks like a good candidate that could be causing this. If we are sure this does not cause unwanted side effects the change looks good.

> If you can find a way to reproduce headers already sent scenario manually

I just saw this once or twice on my end over a period of several months and is was reported by a user. Seems to be highly timing specific, I tried to reproduce it consistently but no luck so far

Reply:
I don't see any side effects for this change.